src/Service/BuscarTodosCarrinhoService.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Service;
+
+use App\Controller\Exception\NaoExistemCarrinhosException;
+use App\Repository\CarrinhoRepository;
+use Exception;
+
+class BuscarTodosCarrinhoService
+{
+  public function __construct(
+    private CarrinhoRepository $carrinhoRepository,
+  ) {}
+
+  public function execute(): array | Exception
+  {
+    $carrinhos = $this->carrinhoRepository->findAll();
+
+    if (!$carrinhos) {
+      return new NaoExistemCarrinhosException();
+    }
+
+    return $carrinhos;
+  }
```

Comment:
exception a gente lança e não retorna, dai ela para o algoritmo...

```suggestion
  /** @return Carrinho[] */
  public function execute(): array
  {
    $carrinhos = $this->carrinhoRepository->findAll();

    if (!$carrinhos) {
      throw new NaoExistemCarrinhosException();
    }

    return $carrinhos;
  }
```